src/contracts/core/DelegationManager.sol

```diff
@@ -661,24 +859,52 @@ contract DelegationManager is
         address operator,
         address withdrawer,
         IStrategy[] memory strategies,
-        uint256[] memory shares
+        uint256[] memory sharesToWithdraw,
+        uint64[] memory totalMagnitudes
     ) internal returns (bytes32) {
         require(
             staker != address(0), "DelegationManager._removeSharesAndQueueWithdrawal: staker cannot be zero address"
         );
         require(strategies.length != 0, "DelegationManager._removeSharesAndQueueWithdrawal: strategies cannot be empty");
 
+        uint256[] memory scaledStakerShares = new uint256[](strategies.length);
+
         // Remove shares from staker and operator
         // Each of these operations fail if we attempt to remove more shares than exist
         for (uint256 i = 0; i < strategies.length;) {
+            // check sharesToWithdraw is valid
+            // TODO maybe have a getter to get totalShares for all strategies, like getDelegatableShares
+            // but for inputted strategies
+            uint256 totalShares;
+            if (strategies[i] == beaconChainETHStrategy) {
+                int256 podShares = eigenPodManager.podOwnerShares(staker);
+                totalShares = podShares <= 0 ? 0 : uint256(podShares);
+            } else {
+                totalShares = strategyManager.stakerStrategyShares(staker, strategies[i]);
+            }
+            require(
+                sharesToWithdraw[i] <= _getWithdrawableShares(staker, strategies[i], totalShares, totalMagnitudes[i]),
+                "DelegationManager._removeSharesAndQueueWithdrawal: shares to withdraw exceeds withdrawable shares"
+            );
+            // calculate scaledShares to place into queue withdrawal and shares to decrement from SM/EPM
+            uint256 stakerScalingFactor = _getStakerScalingFactor(staker, strategies[i]);
+            (
+                // shares to decrement from StrategyManager/EigenPodManager
+                uint256 sharesToDecrement,
+                // scaledShares for staker to place into queueWithdrawal
+                uint256 scaledShares
+            ) = _calculateSharesToQueueWithdraw(sharesToWithdraw[i], stakerScalingFactor, totalMagnitudes[i]);
+            scaledStakerShares[i] = scaledShares;
+
             // Similar to `isDelegated` logic
             if (operator != address(0)) {
                 // forgefmt: disable-next-item
-                _decreaseOperatorShares({
+                _decreaseOperatorScaledShares({
                     operator: operator, 
                     staker: staker, 
                     strategy: strategies[i], 
-                    shares: shares[i]
+                    shares: sharesToWithdraw[i],
```

Comment:
you should just be able to pass `scaledShares` after you change the calculation

Reply:
I guess its duplication of scaled shares logic but the `_decreaseOperatorScaledShares` function is used in other locations (EPM.decreaseDelegatedShares). We can change this implementation once the EPM refactor with the new pod scaling factor idea is being pushed since that will remove the `decreaseDelegatedShares` function entirely.